gyb_syntax_support/Child.py

```diff
@@ -46,16 +46,17 @@ def __init__(self, name, kind, name_for_diagnostics=None, description=None, is_o
         # A restricted set of token kinds that will be accepted for this
         # child.
         self.token_choices = []
-        if self.token:
-            self.token_choices.append(self.token)
+        if self.token and not token_choices:
+            self.token_choices.append((self.token, None))
         for choice in token_choices or []:
+            pipe_index = choice.find('|')
+            choice_text = None
+            if pipe_index != -1:
+                full_choice = choice
+                choice = full_choice[:pipe_index]
+                choice_text = full_choice[(pipe_index+1):]
             token = SYNTAX_TOKEN_MAP[choice]
-            self.token_choices.append(token)
-
-        # A list of valid text for tokens, if specified.
-        # This will force validation logic to check the text passed into the
-        # token against the choices.
-        self.text_choices = text_choices or []
+            self.token_choices.append((token, choice_text))
```

Comment:
Is there a case where `token != KeywordToken` and `kind != KeywordToken`? Can't we just check/use `kind` instead?

Reply:
What exactly are you suggesting? Also, this is going to be removed in a month or so anyway when all gyb files are migrated to CodeGeneration so maybe it doesn’t matter too much anyway.